sympy/matrices/matrices.py

```diff
@@ -2235,6 +2236,9 @@ def strongly_connected_components(self):
     def strongly_connected_components_decomposition(self, lower=True):
         return _strongly_connected_components_decomposition(self, lower=lower)
 
+    def to_domain(self, domain=None):
+        return _to_domain(self, domain=domain)
```

Comment:
The `_to_domain` function is a hack that should not be made public. It is just there as an attempt to maintain backward compatibility for the deprecated way of using the `smith_normal_form` function:
https://github.com/sympy/sympy/blob/ed6a33049b42a16b87a6d8685dd012594aa62dfa/sympy/matrices/tests/test_normalforms.py#L31-L39

There could be a `to_domain` method for `Matrix` but it should not be based on the one `normalforms.py`.

Rather these algorithms should just be implemented for `DomainMatrix` in `sympy/polys/matrices` (like `smith_normal_forms` now is). Then there can be a convenience method for Matrix that calls into the lower-level routine.

See fa6b1f3494d909c43bb589d937b01e72e1f476b2 and the associated PR #21402 for background to why the code is the way that it is.

Reply:
A `to_domain` method would be good but it should just be the function before the changes in fa6b1f3

Reply:
Okay, so I will move my `hermite_normal_form` functions into `polys.matrices.normalforms`.